src/app/qgsmaptoolsplitfeatures.cpp

```diff
@@ -92,44 +92,84 @@ void QgsMapToolSplitFeatures::cadCanvasReleaseEvent( QgsMapMouseEvent *e )
     deleteTempRubberBand();
 
     //bring up dialog if a split was not possible (polygon) or only done once (line)
-    int topologicalEditing = QgsProject::instance()->topologicalEditing();
+    bool topologicalEditing = QgsProject::instance()->topologicalEditing();
+    QgsPointSequence topologyTestPoints;
     vlayer->beginEditCommand( tr( "Features split" ) );
-    QgsGeometry::OperationResult returnCode = vlayer->splitFeatures( captureCurve(), true, topologicalEditing );
-    vlayer->endEditCommand();
-    if ( returnCode == QgsGeometry::OperationResult::NothingHappened )
-    {
-      QgisApp::instance()->messageBar()->pushMessage(
-        tr( "No features were split" ),
-        tr( "If there are selected features, the split tool only applies to those. If you would like to split all features under the split line, clear the selection." ),
-        Qgis::Warning,
-        QgisApp::instance()->messageTimeout() );
-    }
-    else if ( returnCode == QgsGeometry::OperationResult::GeometryEngineError )
-    {
-      QgisApp::instance()->messageBar()->pushMessage(
-        tr( "No feature split done" ),
-        tr( "Cut edges detected. Make sure the line splits features into multiple parts." ),
-        Qgis::Warning,
-        QgisApp::instance()->messageTimeout() );
-    }
-    else if ( returnCode == QgsGeometry::OperationResult::InvalidBaseGeometry )
-    {
-      QgisApp::instance()->messageBar()->pushMessage(
-        tr( "No feature split done" ),
-        tr( "The geometry is invalid. Please repair before trying to split it." ),
-        Qgis::Warning,
-        QgisApp::instance()->messageTimeout() );
-    }
-    else if ( returnCode != QgsGeometry::OperationResult::Success )
+    QgsGeometry::OperationResult returnCode = vlayer->splitFeatures( captureCurve(), topologyTestPoints, false, topologicalEditing );
```

Comment:
Had to change this parameter to `false`. With `true` all split polygons are silently transformed to `CurvePolygon` even if the split was done with a straight line.
@vcloarec would you mind shedding some light on this if it was intentional?

Reply:
Yes, it was intentional. As GEOS doesn't support splitting with curve, there is a workaround that is to discretize the curved elements, split with them and finally de-segmentize split features. But indeed, with that and I missed it, polygons are not polygons anymore but CurvedPolygons, even with straight split... Sorry about that... I purpose to keep the argument true, but to add this in the splitting method to deactivate the preserving if there is no curved element in the QgsCurve argument: PR here : https://github.com/qgis/QGIS/pull/38294

Reply:
thanks!